### PR TITLE
Fix Split chat not working on Livestreams

### LINKED ISSUE #673: chat split not working
### Existing Issues

- [X] I have searched for existing issues, this is not a duplicate.

### Unmodded Twitch

- [X] I have checked and this issue only occurs using bttv-android and not with the unmodded Twitch App

### What happened?

SmartSelect_20240824_012244_BTTV
SmartSelect_20240824_012808_BTTV

### bttv-android version

0.41.0 beta

### android version

13

### device

galaxy tab s7+

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by Zerwin**
Fixes #673 

## Changes
Twitch changed chat messages in Live Streams to be LinearLayouts with the chat_message_item as a child. This fixes it to target the child rather than the LinearLayout to make Split Chat work again.

## Checklist

 - [x] My change builds
 - [x] I tested my change
 - [ ] ~~I use the "bttv_" prefix for all resources I propose~~
 - [ ] ~~When adding a string I also added it to the `bttv.Res.strings` Enum and `res/values/strings.xml` (in `mod`) and `res/values/public.xml` (in `disass`)~~
 - [ ] ~~If my change is significant enough, I added it to the CHANGELOG.md under `master`~~
 - [ ] ~~I'll add myself and everyone else who contributed to this change to the contributors list using [all-contributors](https://allcontributors.org/docs/en/bot/usage)~~
 - [x] I license my changes according to the [MIT License](https://github.com/bttv-android/bttv/blob/master/LICENSE).

### CODE REVIEW

**Comment by FoseFx** [comment]
This only highlights the text, not the whole "row"

image

**Comment by Zerwin** [comment]
The LinearLayout only stretches as far as the text, not the row. I thought of trying to set the width to the whole space, but that might introduce other bugs that could be a headache to deal with.

**Comment by Zerwin** [comment]
I made it so that comments are as wide as the parent element now. This should make the entire "row" colored. I tested it for a bit and at least couldn't immediately find any bugs because of this change.

File: mod/app/src/main/java/bttv/SplitChat.java [hunk 1]
```diff
@@ -59,11 +61,28 @@ public static void setBackgroundColor(int position, RecyclerView.ViewHolder view
         Context context = view.getContext();
 
         // make sure we only change chat message items
+        // Other elements like redeems or Sub Anniversaries are ConstraintLayouts
+
+        // If view is a LinearLayout, check the childrens to find chat_message_item
```
Comment (FoseFx) [inline mod/app/src/main/java/bttv/SplitChat.java:66]:
```suggestion
        // make sure we only change chat message items,
        // which are LinearLayouts with chat_message_item children.
        // Anything else is ignored.
```

File: mod/app/src/main/java/bttv/SplitChat.java [hunk 2]
```diff
@@ -59,11 +61,28 @@ public static void setBackgroundColor(int position, RecyclerView.ViewHolder view
...
+        if(view instanceof LinearLayout) {
+            Log.d(TAG, "view is LinearLayout: " + view.toString());
+            LinearLayout linearLayout = (LinearLayout) view;
+            for (int j = 0; j < linearLayout.getChildCount(); j++) {
+                View nestedChild = linearLayout.getChildAt(j);
+                if (nestedChild.getId() == ResUtil.getResourceId(context, "chat_message_item", "id")) {
```
Comment (FoseFx) [inline mod/app/src/main/java/bttv/SplitChat.java:72]: before the loop (before the `if` even), initialize a variable `chatMessageItemResId` with `ResUtil.getResourceId(context, "chat_message_item", "id")`. This way it does not have to be looked up in every iteration.

Also do `chommentRootViewResId` for consistency.

```suggestion
                if (nestedChild.getId() == chatMessageItemResId) {
```

File: mod/app/src/main/java/bttv/SplitChat.java [hunk 3]
```diff
@@ -59,11 +61,28 @@ public static void setBackgroundColor(int position, RecyclerView.ViewHolder view
...
+                    Log.d(TAG, "found chat_message_item: " + nestedChild.toString());
+                    view = nestedChild;
+                    // Increase width of linearLayout to match parent to color the whole item
+                    linearLayout.setLayoutParams(new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT));
+                    break;
+                }
+            }
+        }
         boolean hasChatMessageId = view.getId() == ResUtil.getResourceId(context, "chat_message_item", "id");
```
Comment (FoseFx) [inline mod/app/src/main/java/bttv/SplitChat.java:81]:
```suggestion
        boolean hasChatMessageId = view.getId() == chatMessageItemResId;
```

File: mod/app/src/main/java/bttv/SplitChat.java [hunk 4]
```diff
@@ -59,11 +61,28 @@ public static void setBackgroundColor(int position, RecyclerView.ViewHolder view
...
         boolean hasChommentRootId = view.getId() == ResUtil.getResourceId(context, "chomment_root_view", "id");
```
Comment (FoseFx) [inline mod/app/src/main/java/bttv/SplitChat.java:82]:
```suggestion
        boolean hasChommentRootId = view.getId() == chommentRootViewResId;
```

File: mod/app/src/main/java/bttv/SplitChat.java [hunk 5]
```diff
@@ -59,11 +61,28 @@ public static void setBackgroundColor(int position, RecyclerView.ViewHolder view
...
 
-        if (!hasChatMessageId && !hasChommentRootId) {
-            Log.i(TAG, "view skipped, as it's not a chat message or chomment" + viewHolder.toString());
+        if (!(hasChatMessageId || hasChommentRootId)) {
+            Log.i(TAG, "view skipped, as it's not a chat message or chomment, " + viewHolder.toString() + " ID: " + view.getId() + " View: " + view.toString() + " Expected ID: " + ResUtil.getResourceId(context, "chat_message_item", "id") + " or " + ResUtil.getResourceId(context, "chomment_root_view", "id"));
```
Comment (FoseFx) [inline mod/app/src/main/java/bttv/SplitChat.java:85]:
```suggestion
            Log.d(TAG, "view skipped, as it's not a chat message or chomment, " + viewHolder.toString() + " ID: " + view.getId() + " View: " + view.toString() + " Expected ID: " + chatMessageItemResId + " or " + chommentRootViewResId);
```

**Review by FoseFx — CHANGES_REQUESTED** [review]
Highly appreciate your help!
Only one minor thing I'd like to have changed with this.

**Comment by Zerwin** [comment]
This should be all of the changes

**Review by FoseFx — APPROVED** [review]
🎉️

**Pull request merged**